ls1intum/Artemis
# [Enhancement]Student Questions & Answers: Security Improvements

### Checklist

### Motivation and Context



Follow-up PR to issues discussed in https://github.com/ls1intum/Artemis/pull/2156.

### Description


Added security checks if the id of the course provided by the url matches the id of the course the question or answer is in.

### Steps for Testing


1. Log in to Artemis
2. Check that there is a course with exercises and lectures, and questions are enabled.
3. Check that all functionalities for both questions and answers still work as expected. (https://ls1intum.testquality.com/project/10165/plan/17385/suite/80878/testcase/234270 might help to test all existing functionality)

### Test Coverage

* StudentQuestionResource.java: 83%
* StudentQuestionAnswerResource.java 88%
